base/combinatorics.jl

```diff
@@ -14,9 +14,11 @@ for n in 2:34
     _fact_table128[n] = _fact_table128[n-1] * n
 end
 
-function factorial_lookup(n::Integer, table, lim)
+@assume_effects :consistent :noub function factorial_lookup(
```

Comment:
rather than annotating consistent, you could make the table be a Tuple (see exp.jl for the best way I've found to do this type of thing)